index.js

```diff
@@ -1483,14 +1489,101 @@ class Hydra extends EventEmitter {
   }
 
   /**
-   * @name _getRandomServicePort
-   * @summary Retrieves a random TCP/IP port.
-   * @return {number} port - new random socket port
+   * @name _parseServicePortConfig
+   * @summary Parse and process given port data in config
+   * @param {mixed} port - configured port
+   * @return {promise} promise - resolving with unassigned port, rejecting when no free port is found
    */
-  _getRandomServicePort() {
-    const maxSocketPort = 65535;
-    const nonPriviliagePortBountry = 1024;
-    return parseInt(nonPriviliagePortBountry + (new Date().getTime() % (Math.random() * (maxSocketPort - nonPriviliagePortBountry))));
+  _parseServicePortConfig(port) {
+    //No port given, get unassigned port from standard ranges
+    if (typeof port === 'undefined' || !port || port == 0) {
```

Comment:
Maybe there's a misreading, because your example shows typedef? typeof is an operator in JS since the early versions. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof

> // Undefined
> typeof undefined === 'undefined';
> typeof declaredButUndefinedVariable === 'undefined';
> typeof undeclaredVariable === 'undefined';